<robot name="single_joint_example">

    <link name="base_link">
        <inertial>
            <mass value="100" />
            <inertia ixx="100" ixy="0" ixz="0" iyy="100" iyz="0" izz="100" />
        </inertial>
        <visual>
            <geometry>
                <box size="0.3 0.3 1" />
            </geometry>
        </visual>
        <collision>
            <geometry>
                <box size="1 1 1" />
            </geometry>
        </collision>
    </link>

    <joint name="axis" type="continuous">
        <origin xyz="0 0 0.55" rpy="0 0 0" />
        <parent link="base_link" />
        <child link="child_link" />
        <axis xyz="0 0 1" />
        <limit effort="100." velocity="1." />
    </joint>

    <link name="child_link">
        <inertial>
            <mass value="1" />
            <inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1" />
        </inertial>
        <visual>
            <!--
            <geometry>
                <box size="1 1 0.1" />
            </geometry>
            -->
            <geometry>
                <mesh filename="package://root/layered_hw_ws/src/layered_hardware_ign/urdf/child_link.stl" scale="0.01 0.01 0.01"/>
            </geometry>
        </visual>
        <collision>
            <geometry>
                <box size="1 1 0.1" />
            </geometry>
        </collision>
    </link>

    <ros2_control name="LayeredHardwareIgnition" type="system">
        <hardware>
            <plugin>layered_hardware_ign/LayeredHardwareIgnition</plugin>
            <param name="layers">
                - name: monitor_layer
                  type: layered_hardware/MonitorLayer
                - name: joint_limits_layer
                  type: layered_hardware/JointLimitsLayer
                - name: ign_joint_layer
                  type: layered_hardware_ign/IgnitionJointLayer
            </param>
            <param name="ign_joint_layer">
                joints:
                    axis:
                        operation_mode_map:
                            axis/position: position
                            axis/velocity: velocity
                            axis/effort: effort
            </param>
        </hardware>
        <joint name="axis">
            <command_interface name="position">
                <param name="min">-1</param>
                <param name="max">1</param>
            </command_interface>
            <command_interface name="velocity">
            </command_interface>
            <command_interface name="effort">
            </command_interface>
            <state_interface name="position"/>
        </joint>
    </ros2_control>

    <gazebo>
        <plugin filename="ign_ros2_control-system" name="ign_ros2_control::IgnitionROS2ControlPlugin">
            <parameters>$(find layered_hardware_ign)/config/example_controllers.yaml</parameters>
        </plugin>
    </gazebo>

</robot>